src/parser/syntax.rs

```diff
@@ -85,61 +89,116 @@ pub fn parse_symbol_limbs<F: LurkField>() -> impl Fn(Span<'_>) -> ParseResult<'_
     }
 }
 
-pub fn parse_absolute_symbol<F: LurkField>() -> impl Fn(Span<'_>) -> ParseResult<'_, F, Symbol> {
+fn intern_path<'a, F: LurkField>(
+    state: Rc<RefCell<State>>,
```

Comment:
Does this need more than `&State` ? Same question for its callers.

Reply:
It needs a mutable state because `intern` needs a mutable state

Reply:
Yes, but mutability accounts for`&mut State`, not `Rc<RefCell<...>>`. The later is only required because of the preparation of a sequence of mutable references to state in the parser combinator alt used in `parse_symbol`:
https://github.com/lurk-lab/lurk-rs/pull/578/files#diff-ad4481d9d6d12a81307a7986448cf9d889e0249c193c956d38dc2bf896a12424R190-R199

The current solution is fine for now, I guess. There are methods called by `parse_symbol` that do not need the `Rc`, but that's a detail at this stage.